### ISSUE TITLE
Parameter get renamed in compound model

### ISSUE THREAD

**Issue by wkerzendorf** [astropy/astropy#3768]
Presumably for @embray or @nden: If I make a composition of two models which have a parameter `a` and parameter `b` respectively. These get then translated into `a_0` and `b_1`. Is there any way to avoid this.

**Comment by embray** [comment]
That's correct--the parameters on the compound models are given new names that make it clear which sub-model they're associated with.  There is not currently any way to give custom names to the parameters in a compound model.  There *should* be, but it's not clear what the best approach to that is.

**Comment by wkerzendorf** [comment]
Is there a question about the API? My suggestion would be to have them not change their name unless they are duplicate - but that's from my point of view. I think we need to build use cases to see what approach is needed. So for now it would be already great if I can just rename parameters myself. Then we can see if what my API is sensible in other places. Is there a way to rename parameters?

**Comment by embray** [comment]
> My suggestion would be to have them not change their name unless they are duplicate - but that's from my point of view.

That's the most obvious alternative in terms of automatic renaming.  But making a compound model with two of the same model (example adding two Gaussians) is quite common.  It can also quickly become confusing with more than two models which parameters go with which, and even worse still if you combine two compound models together.  So one could come up with different rules that work better in different contexts, I agree.  But at the end of the day it's least confusing to have a single rule for all cases.

That said, there's no doubt that there needs to be a way to give parameters new names in a compound model.  There also needs to be a way to fix some parameters to a constant, so that they are effectively dropped from the description of the model.  Similarly, there is a need to combine redundant parameters (I could swear there was an issue for the latter, but I can't find it; maybe it was only discussed on the mailing list).

**Comment by wkerzendorf** [comment]
Well, I mean if you have duplicate names you will start making it `a_2` if it doesn't then you leave as is. I agree that this is not quite clear what way to pursue. just give me one way to change parameter names and i'll implement one way in my code and we can see if we like it. Can I hack together some API for me to change the parameter name?

**Comment by embray** [comment]
In thinking about this, I've been cooking up a few ideas.  There are two levels at which one might want to handle this.  One is at the class level--for example, it should be possible to subclass an existing model and give new names to the existing parameters on that model.

At another level, it may be very useful to rename parameters at a per-instance level.  In both cases this would require providing a mapping (call it parameter_aliases, say) mapping default parameter names to any new names you might want to give for that parameter.

**Comment by wkerzendorf** [comment]
Something like that would be useful. You also mentioned something like `fixed` before - setting parameters to constant during fitting. I essentially attached a fit_evaluate function to a compound model that would replace all non-fixed parameters with the new input and then use the evaluate function.

One last thing is there a fix yet for `inputs = ()` - sorry it's sort of important for starkit.
